pkg/wasm/cache.go

```diff
@@ -160,19 +156,34 @@ func (c *LocalFileCache) Get(downloadURL, checksum string, timeout time.Duration
 		}
 		wasmLog.Debugf("wasm oci fetch %s with options: %v", downloadURL, imgFetcherOps)
 		fetcher := NewImageFetcher(ctx, imgFetcherOps)
-		b, dChecksum, err = fetcher.Fetch(u.Host+u.Path, checksum)
+		binaryGetter, dChecksum, err = fetcher.Fetch(u.Host + u.Path)
```

Comment:
a function named `Fetch` but actually you may done get binary on L180, how about add a new function check the cache, WDYT?

Reply:
I think it's ok to let `HttpFetcher` different with `ImageFetcher`.